### DISCUSSION TITLE
[Feature] Search by video duration, file checksum, or filesize

### DISCUSSION THREAD

**Discussion by mmomjian** [immich-app/immich#9139] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Feature request: add a search filter to search by video duration > or < than a certain length. Also add a filter to search by file checksum or filesize (< or >), as we currently advise users to query the database for this. All of this data is already stored in the database (`assets.duration`, `assets.checksum`, `exif.fileSizeInByte`) so I believe it should be possible to implement.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by hexxone** [comment]
+1 For this feature.

I would also like to be able to sort by video duration if possible.
Because I have several "long format" videos which are basically "super-cuts" of some old photo and Video DVDs.

Currently I can only easily find them by navigating through the related album.
Sorting by file size would not really help, because they greatly differ in size.
E.g.: some 10min GoPro video from 2023 is probably larger than most of the 10year old 2-hour long DVDs.

**Comment by agjmills** [comment]
+1 for this feature

**Comment by The-Judge** [comment]
Also +1 from me!

I'd like to add, that it would be especially helpful, if this search by duration / dimensions would be combined with a "duplicate score", so it get's easier to identify Low-Res versions from existing or about-to-upload High-Res variants of the same content.
This happens regularly when uploading WhatsApp files of local high-res Images taken by a smartphone, since WhatsApp and similar apps resize images before they are sent.

**Reply by seb-1204** [reply]
gosh, videos shared via WhatsApp or similar are prime candidate for this. Same for photos.

**Comment by pustekuchen91** [comment]
+1 

use case: find meme images (mostly just some kB) and delete them.

**Reply by Meesmaalt** [reply]
+1 Would like video duration sorting!

**Reply by webysther** [reply]
or just a checkbox with short video

**Comment by Chefdebar** [comment]
Would love this option. I have some really small duplicate files of photos that were created. They are under 10 KiB and i need a way to search to delete them.

**Comment by stantheblan** [comment]
!!! BUMP !!!
